packages/deck-scripts/config/basePluginRollupConfig.js

```diff
@@ -0,0 +1,55 @@
+const baseRollupConfig = require('./baseRollupConfig');
+const externalGlobals = require('rollup-plugin-external-globals');
+
+module.exports = {
+  ...baseRollupConfig,
+  input: 'src/index.ts',
+  output: [{ dir: 'build/dist', format: 'es', sourcemap: true }],
+  plugins: [
+    ...baseRollupConfig.plugins,
+    // map imports from shared libraries (react, etc) to global variables exposed by spinnaker
+    externalGlobals(spinnakerSharedLibraries()),
+  ],
+};
+
+function spinnakerSharedLibraries() {
+  // Updates here should also be added in core/src/plugins/sharedLibraries.ts
+  const libraries = [
+    '@spinnaker/core',
+    '@uirouter/core',
+    '@uirouter/react',
+    '@uirouter/rx',
+    'lodash',
+    'prop-types',
+    'react',
+    'react-dom',
+    'rxjs',
+    'rxjs/Observable',
+  ];
+
+  function getGlobalVariable(libraryName) {
+    const prefix = 'spinnaker.plugins.sharedLibraries';
+    const sanitizedLibraryName = libraryName.replace(/[^a-zA-Z0-9_]/g, '_');
+    return `${prefix}.${sanitizedLibraryName}`;
+  }
+
+  // This allows us to share libraries with plugin dependencies that would otherwise be double bundled
+  // @rollup/plugin-commonjs rewrites library names in commonjs imported code
+  // This block finds known permutations and replaces them with the shared global variable
+  return libraries.reduce((globalsMap, libraryName) => {
+    const globalVar = getGlobalVariable(libraryName);
+    const libName2 = libraryName + '?commonjs-proxy';
+    const libName3 = libraryName + '?commonjs-require';
+    const libName4 = '\u0000' + libraryName + '?commonjs-proxy';
+    const libName5 = '\u0000' + libraryName + '?commonjs-require';
```

Comment:
Not sure that I understand this (even after reading the comment above). Can you elaborate more?

Reply:
What's the purpose of `\u0000`?

Reply:
the null character is a rollup-ism that marks a module created within a plugin as "synthetic".  In this case, the `@rollup/plugin-commonjs` uses it as a marker for the commonjs compatibility modules it creates under the covers.  

I'm not confident that processing all these permutations is necessary for the plugin's rollup config to work properly, but I think at least some of them are necessary.